chia/wallet/wallet_node.py

```diff
@@ -355,7 +356,7 @@ async def _start_with_fingerprint(
         path.parent.mkdir(parents=True, exist_ok=True)
         if self.config.get("reset_sync_for_fingerprint") == fingerprint:
             await self.reset_sync_db(path, fingerprint)
-
+        self.log_in(private_key)
```

Comment:
We need to know the logged-in fingerprint when we create the wallet state manage for fetching fingerprint level config.

Reply:
Requiring log in before instantiating wallet state manager seems like a pretty big change, could you change it so that you just get `WalletNode.logged_in_fingerprint` when you need it and throw an error if for some reason you're not logged in?

Reply:
It is needed during the wallet state manager initialization. We can do a lazy load trick but we are adding more fingerprint-level configs, e.g. the address book. Some of them may not be able to lazy load. What is the known concern for logging before the wallet manager initialization?

Reply:
Given we seem to use the `WalletNode.logged_in` and `WalletNode.logged_in_fingerprint` as flags to decide wether loading is done it seems better to not "log in" before we are done and to just get `puzzle_decorator` from the config upfront and pass `puzzle_decorator` to `WalletStateManager.create`? But more important question, why does this need to be based on the fingerprint in the config? Seems strange given that the fingerprint is not really deterministic.